### PR TITLE
Fix opset version of the optimizer in function generate_artifacts

### PR DESCRIPTION
**Pull request by xadupre**
### Description
`generate_artifacts` generates 4 graphs for training. All graphs should share the same opset version, the one coming from the model to train, but the optimizer is left undefined. onnxruntime is using the latest version defined by onnx but onnxruntime does not necessarily support it.

### Motivation and Context
The code does not let the user change it.

### CODE REVIEW

File: orttraining/orttraining/python/training/artifacts.py [hunk 1]
```diff
@@ -53,6 +53,8 @@ def generate_artifacts(
         3. Checkpoint (directory): Contains the model parameters.
         4. Optimizer model (onnx.ModelProto): Model containing the optimizer graph.
 
+    All generated ModelProto are using the same opsets defined by *model*.
```
Comment (askhade) [inline orttraining/orttraining/python/training/artifacts.py:56]:
```suggestion
    All generated ModelProtos will use the same opsets defined by *model*.
```

File: orttraining/orttraining/python/training/artifacts.py [hunk 2]
```diff
@@ -207,11 +209,17 @@ def _export_to_ort_format(model_path, output_dir, extra_options):
 
     logging.info("Optimizer enum provided: %s", optimizer.name)
 
+    opset_version = None
+    for domain in model.opset_import:
+        if domain.domain == "":
```
Comment (askhade) [inline orttraining/orttraining/python/training/artifacts.py:214]: Should this be domain.domain == "" or domain.domain=="ai.onnx" ?

**Comment by askhade** [comment]
Can you add a test to check all the models are generated with the same opset version as the original supplied model. All the tests for onnxblock are here: "https://github.com/microsoft/onnxruntime/blob/main/orttraining/orttraining/test/python/orttraining_test_onnxblock.py"

File: orttraining/orttraining/python/training/artifacts.py (hunk 2 above)
Comment (xadupre) [reply orttraining/orttraining/python/training/artifacts.py:214]: I'll add it but I never saw ai.onnx ever used.

**Review by askhade — APPROVED** [review]

**Pull request merged**